.github/workflows/conformance-eks.yaml

```diff
@@ -251,48 +263,23 @@ jobs:
 
       - name: Run connectivity test (${{ join(matrix.*, ', ') }})
         run: |
-          cilium connectivity test ${{ steps.vars.outputs.connectivity_test_defaults }} \
+          ./cilium-cli connectivity test ${{ steps.vars.outputs.connectivity_test_defaults }} \
           --junit-file "cilium-junits/${{ env.job_name }} (${{ join(matrix.*, ', ') }}) - 1.xml" \
           --junit-property github_job_step="Run connectivity test (${{ join(matrix.*, ', ') }})"
 
-      - name: Clean up Cilium
-        run: |
-          pkill -f "cilium.*hubble.*port-forward|kubectl.*port-forward.*hubble-relay" || test $? -eq 1
-          cilium uninstall --wait
-
-      - name: Create custom IPsec secret
-        run: |
-          kubectl create -n kube-system secret generic cilium-ipsec-keys --from-literal=keys="15 rfc4106(gcm(aes)) $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null | xxd -p -c 64)) 128"
```

Comment:
There's a subtle, undocumented functional diff here in that you changed the tested SPI. Offline, you mentioned you had an issue with key rotations from 15 to 1. Was that a false positive or do we still need to track it?

Reply:
Added:
```
    In addition, this commit fixes the key rotation test code bug.
    Previously, when determing a keyid before the rotation, we ran "cut -c
    1". This was returning only the first digit (e.g., if keyid was "15",
    then "1" was returned). This was breaking the rotation 15=>1.
```

Reply:
Fix and refactoring mixed in the same commit :grimacing: No wonder I missed it.

Reply:
@brb @pchaigno is it feasible to split refactor and fix?

Reply:
Added https://github.com/cilium/cilium/pull/29704/commits/140141e6e34c1ba216c0ddeb2a2f8de94f7892f7.